app/src/main/java/com/heig/atmanager/stats/StatsFragment.java

```diff
@@ -85,144 +90,244 @@ public void onNothingSelected(AdapterView<?> parent) {
             }
         });
 
-
-        //Charts
-        //user = ((MainActivity) getContext()).getUser();
-
-        //TODO : get this by values.colors, not working for some reason
-        bgColor = "#F1F1F1";
-
-        pieChartTasks = v.findViewById(R.id.pie_chart_tasks);
-        pieChartTasks.setBackgroundColor(bgColor);
-        lineChartTasks = v.findViewById(R.id.line_chart_tasks);
-        lineChartTasks.setProgressBar(v.findViewById(R.id.progress_bar));
-        lineChartTasks.setBackgroundColor(bgColor);
-        pieChartGoals = v.findViewById(R.id.pie_chart_goals);
-        pieChartGoals.setBackgroundColor(bgColor);
-
-        //makeCharts(Interval.DAY);
+        //init user and charts
+        user = MainActivity.getUser();
+        bgColor = "#" + getResources().getString(R.color.colorPrimary).substring(3);
+        initCharts(v);
 
         return v;
     }
 
-    /*private void makeCharts(Interval interval){
-
-        //TODO : Select goals/Tasks based on Interval
-        tasks = user.getTaskLists().getValue().get(0).getTasks();
-        goals = user.getGoals().getValue();
-
-        if(tasks != null) { //TODO : Something with no tasks
-            APIlib.getInstance().setActiveAnyChartView(pieChartTasks);
-            makePieChartTasks();
-            APIlib.getInstance().setActiveAnyChartView(lineChartTasks);
-            makeLineChartTasks(interval);
+    /**
+     * Rafraichi les graphes
+     * @param interval temps
+     */
+    private void makeCharts(Interval interval){
+
+        String lineChartLegend = "";
+
+        //keep interval tasks
+        switch(interval){
+            case DAY:
+                tasks = user.getTasksForDay(new Date());
+                goals = user.getDailyGoalTodo();
+                lineChartLegend = "Hours";
+                break;
+            case WEEK:
+                tasks = user.getTasksForLastWeek();
+                goals = user.getWeeklyGoalTodo();
+                lineChartLegend = "Days";
+                break;
+            case MONTH:
+                tasks = user.getTasksForLastMonth();
+                goals = user.getMonthlyGoalTodo();
+                lineChartLegend = "Days";
+                break;
+            case YEAR:
+                tasks = user.getTasksForLastYear();
+                goals = user.getYearlyGoalTodo();
+                lineChartLegend = "Months";
+                break;
```

Comment:
I think you should use the values of the enum (noun), and then capitalize + add an s
=> lineChartLegend = capitalize(interval.getNoun) + "s";

Reply:
It is not according to interval so no